### PR TITLE
feat: Match JSON tags from document

### LINKED ISSUE #189: JSON Tags
Match json tags from the spec

### PR DESCRIPTION
**Pull request by sandrask**
Updated JSON tags, renamed model and vars to match tags: patch data to delta, did unique suffix to did suffix etc.

Closes #189

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

File: pkg/api/batch/operation.go [hunk 1]
```diff
@@ -28,14 +28,14 @@ type Operation struct {
...
+	// operation delta
+	Delta *model.DeltaModel `json:"delta"`
 
 	// suffix data
 	SuffixData *model.SuffixDataModel `json:"suffixData"`
 
-	// encoded patch data
-	EncodedPatchData string `json:"encodedPatchData"`
+	// encoded delta
+	EncodedDelta string `json:"encodedDelta"`
```
Comment (troyronda) [inline pkg/api/batch/operation.go:38]: they use underscores rather than snake case.

File: pkg/api/batch/operation.go [hunk 2]
```diff
@@ -52,9 +52,9 @@ type Operation struct {
 	RecoveryRevealValue string `json:"recoveryRevealValue"`
 
 	// Hash of reveal value for the next update operation
-	NextUpdateCommitmentHash string `json:"nextUpdateCommitmentHash"`
+	UpdateCommitment string `json:"updateCommitment"`
```
Comment (troyronda) [inline pkg/api/batch/operation.go:55]: underscores

File: pkg/api/batch/operation.go [hunk 3]
```diff
@@ -52,9 +52,9 @@ type Operation struct {
...
 	// Hash of reveal value for next recovery/deactivate operation
-	NextRecoveryCommitmentHash string `json:"nextRecoveryCommitmentHash"`
+	RecoveryCommitment string `json:"recoveryCommitment"`
```
Comment (troyronda) [inline pkg/api/batch/operation.go:57]: underscores

Comment (sandrask) [reply pkg/api/batch/operation.go:57]: Operation is our internal model - so we can use whatever we want.

File: pkg/api/batch/operation.go (hunk 2 above)
Comment (sandrask) [reply pkg/api/batch/operation.go:55]: Operation is our internal model

File: pkg/api/batch/operation.go (hunk 1 above)
Comment (sandrask) [reply pkg/api/batch/operation.go:38]: Operation is our internal model

File: pkg/api/batch/operation.go (hunk 3 above)
Comment (troyronda) [reply pkg/api/batch/operation.go:57]: Prefer to be consistent. I'll open a fresh issue for the internal model.

Comment (troyronda) [reply pkg/api/batch/operation.go:57]: https://github.com/trustbloc/sidetree-core-go/issues/221

**Pull request merged**